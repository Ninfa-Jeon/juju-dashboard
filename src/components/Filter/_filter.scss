@import "vanilla-framework/scss/settings_colors";

.p-filter {
  &__label {
    color: $color-mid-dark;
  }

  &__label,
  &__item {
    margin-right: 0.5rem;
  }
}
