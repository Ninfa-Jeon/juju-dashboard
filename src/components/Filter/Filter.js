import React, { Component } from "react";
import Button from "../Button/Button";

import "./_filter.scss";

export default class Filter extends Component {
  render() {
    const { label, filters } = this.props;
    const filterButtons = filters.map(filter => (
      <span key={filter} className="p-filter__item">
        <Button>{filter}</Button>
      </span>
    ));
    return (
      <div className="p-filter">
        <span className="p-filter__label">{label}</span>
        {filterButtons}
      </div>
    );
  }
}
