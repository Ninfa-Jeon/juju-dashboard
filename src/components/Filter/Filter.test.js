import React from "react";
import { shallow } from "enzyme";

import Filter from "./Filter";

describe("Filter", () => {
  it("renders without crashing and matches snapshot", () => {
    const filters = ["foo", "bar", "baz", "foobar"];
    const wrapper = shallow(<Filter label="Filter:" filters={filters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
