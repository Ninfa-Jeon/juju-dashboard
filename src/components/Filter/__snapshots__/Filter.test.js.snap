// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders without crashing and matches snapshot 1`] = `
<div
  className="p-filter"
>
  <span
    className="p-filter__label"
  >
    Filter:
  </span>
  <span
    className="p-filter__item"
    key="foo"
  >
    <Button>
      foo
    </Button>
  </span>
  <span
    className="p-filter__item"
    key="bar"
  >
    <Button>
      bar
    </Button>
  </span>
  <span
    className="p-filter__item"
    key="baz"
  >
    <Button>
      baz
    </Button>
  </span>
  <span
    className="p-filter__item"
    key="foobar"
  >
    <Button>
      foobar
    </Button>
  </span>
</div>
`;
